Week1/homework/app.js

```diff
@@ -1,11 +1,149 @@
 'use strict';
 
-{
-  const bookTitles = [
-    // Replace with your own book titles
-    'harry_potter_chamber_secrets'
-  ];
-
-  // Replace with your own code
-  console.log(bookTitles);
+// 1.1
+const bookTitles = [
+  // Replace with your own book titles
+  'fairy_tales',
+  'things_fall_apart',
+  'the_divine_comedy',
+  'the_epic_of_gilgamesh',
+  'the_book_of_job',
+  'one_thousand_and_one_nights',
+  'pride_and_prejudice',
+  'the_decameron',
+  'ficciones',
+  'the_stranger'
+];
+
+// Replace with your own code
+// console.log(bookTitles);
+
+// 1.3
+// const ulEle = document.createElement('ul');
+
+// function renderTitleBook(ulList) {
+//   for (const title of bookTitles) {
+//     const liElement = document.createElement('li');
+//     liElement.innerText = title;
+//     ulList.appendChild(liElement);
+//   }
+// }
+
+// // console.log(ul)
+// renderTitleBook(ulEle);
+
+// 1.4
+const titleProperties = {
+  things_fall_apart: {
+    title: 'Things Fall Apart',
+    language: 'English',
+    author: 'Chinua Achebe'
+  },
+  fairy_tales: {
+    title: 'Fairy tales',
+    language: 'Denmark',
+    author: 'Hans Christian Andersen'
+  },
+  the_divine_comedy: {
+    title: 'The Divine Comedy',
+    language: 'Italy',
+    author: 'Dante Alighieri'
+  },
+  the_epic_of_gilgamesh: {
+    title: 'The Epic Of Gilgamesh',
+    language: 'Akkadian',
+    author: 'Unknown'
+  },
+  the_book_of_job: {
+    title: 'The Book Of Job',
+    language: 'Hebrew',
+    author: 'Wuthering Heights'
+  },
+  one_thousand_and_one_nights: {
+    title: 'One Thousand and One Nights',
+    language: 'Arabic',
+    author: 'Samuel Beckett'
+  },
+  pride_and_prejudice: {
+    title: 'Pride and Prejudice',
+    language: 'English',
+    author: 'Jane Austen'
+  },
+  the_decameron: {
+    title: 'The Decameron',
+    language: 'Italian',
+    author: 'Giovanni Boccaccio'
+  },
+  ficciones: {
+    title: 'Ficciones',
+    language: 'Spanish',
+    author: 'Jorge Luis Borges'
+  },
+  the_stranger: {
+    title: 'The-Stranger',
+    language: 'French',
+    author: 'Albert Camus'
+  }
+};
+
+// Create append function
+function modifyAppendChildFunc(parent = 'document.body', child) {
```

Comment:
Setting a default value for a parameter is only useful at the end of the parameter list. If I want to use the default parameter as you have used it I would need to specify `undefined` for that parameter, e.g.:

```
modifyAppendChildFunc(undefined, headerElement);
```

So what would be the point?

But apart from that, nowhere in the rest of your code your are making use of this 'default' value. Therefore it is just noise.

With regard to what this function does: it only appends a child to a parent. But that is exactly what `parent.appendChild(...)` already does. So what is the point of this function?